lib/fruitbot/streampusher_api.ex

```diff
@@ -1,35 +1,68 @@
 defmodule Fruitbot.StreampusherApi do
   def user_search(query) do
-  end
+    case HTTPoison.get!("https://datafruits.streampusher.com/api/djs/#{query}.json?name=#{query}") do
+      %HTTPoison.Response{status_code: 200, body: body} ->
+        response = Jason.decode!(body)
+        data = response["data"]
+        username = data["attributes"]["username"]
+        url = "https://datafruits.fm/djs/#{username}"
+        "Datafruit found: #{url}"
+      %HTTPoison.Response{status_code: 404} ->
+        "#{query} not found"
 
-  def tag_search(query) do
+      # elixis LS for some reason tells me an error won't occur, dunno why, but we can just use a catch-all for any response besides success
+      _ ->
+        "Whoops must have eaten a bad fruit"
+
+        # %HTTPoison.Error{reason: reason} ->
+      #   IO.inspect(reason)
+    end
   end
 
-  def wiki_search(query) do
-    response =
-      case HTTPoison.get!("https://datafruits.streampusher.com/api/wiki_pages.json?q=#{query}") do
-        %HTTPoison.Response{status_code: 200, body: body} ->
-          Jason.decode!(body)
+  def tag_search(query) do
+    case HTTPoison.get!("https://datafruits.streampusher.com/api/archives.json?tags=#{query}") do
+      %HTTPoison.Response{status_code: 200, body: body} ->
+        response = Jason.decode!(body)
+        data = response["data"]
+        first_result = Enum.at(data, 0)
 
-        # not sure it seems the API doesn't return a 404 when search result not found, just an empty list
-        %HTTPoison.Response{status_code: 404} ->
+          # uhhh there aren't single pages for achives yet so I'm not sure what link to return...
```

Comment:
yes i'm seeing that. we can grab a random entry and return a streaming link? or return the artwork thumbnail in the chat? all tags associated with shows, yes?

Reply:
Think we can return the direct mp3 link for now

Reply:
ok leave it to me jefe

Reply:
Cool I'll merge this PR , then you can make another PR to add that if you'd like

Reply:
excellent!